src/UrlParsers/SitemapUrlParser.php

```diff
@@ -74,7 +74,7 @@ protected function normalizeUrl(UriInterface $url): UriInterface
 
     protected function shouldCrawl(Node $node): bool
     {
-        if ($this->crawler->mustRespectRobots() && ! $this->crawler->getRobotsTxt()->allows($node->getValue(), $this->crawler->getUserAgent())) {
+        if ($this->crawler->mustRespectRobots() && null !== $this->crawler->getRobotsTxt() && ! $this->crawler->getRobotsTxt()->allows($node->getValue(), $this->crawler->getUserAgent())) {
```

Comment:
Could you break this condition up, use early returns, and avoid `&&` and `||`

Reply:
Done :+1: